app/models/tasks/copy_image_task.rb

```diff
@@ -1,46 +1,56 @@
 class Tasks::CopyImageTask < ::Task
-  before_save do
-    if image_resource && status_changed?
-      image_resource.status = status
+  before_save :update_image, if: ->(t) { t.status_changed? && t.image }
 
-      if complete?
-        image_resource.assign_attributes(porter_callback_image_meta)
-        image_resource.status = "invalid" if image_resource.invalid?
-      end
-
-      image_resource.save!
-      podcast.try(:publish!) if image_resource.status_complete?
-    end
+  def image
+    owner
   end
 
-  def task_options
-    super.merge({
-      job_type: "image",
-      source: image_resource.href,
-      destination: destination_url(image_resource)
-    }).with_indifferent_access
+  def podcast
+    image&.episode&.podcast || image&.podcast
   end
 
-  def destination_url(image_resource)
-    URI::Generic.build(
-      scheme: "s3",
-      host: feeder_storage_bucket,
-      path: image_path(image_resource)
-    ).to_s
+  def source_url
+    image.href
   end
 
-  def image_path(image_resource)
-    if !image_resource
-      logger.info("in CopyImageTask#image_path and image_resource is nil. self is #{inspect}")
-    end
-    URI.parse(image_resource.published_url).path
+  def porter_tasks
+    [
+      {
+        Type: "Inspect"
+      },
+      {
+        Type: "Copy",
+        Mode: "AWS/S3",
+        BucketName: ENV["FEEDER_STORAGE_BUCKET"],
+        ObjectKey: porter_escape(image.path),
+        ContentType: "REPLACE",
+        Parameters: {
+          CacheControl: "max-age=86400",
+          ContentDisposition: "attachment; filename=\"#{porter_escape(image.file_name)}\""
+        }
+      }
+    ]
   end
 
-  def podcast
-    image_resource.try(:episode).try(:podcast) || image_resource.try(:podcast)
-  end
+  def update_image
+    image.status = status
 
-  def image_resource
-    owner
+    if complete?
```

Comment:
This code came from (1) the `before_save` block that was at the top of this file, and (2) the "parse callback metadata" logic from the old `PorterDecoder` concern.